tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -460,5 +497,169 @@ private static bool ParseBooleanConfigurationValue(string value)
+
+                                if (subKey?.GetValue("DisplayName") is datadog64BitProgram or datadog32BitProgram && displayName is null)
+                                {
+                                    displayName = (string?)subKey.GetValue("DisplayName");
+                                    tracerVersion = subKey.GetValue("VersionMajor") + "." + subKey.GetValue("VersionMinor");
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (displayName is null)
+                {
+                    Utils.WriteError(TraceProgramNotFound);
+                }
+                else if (displayName is datadog64BitProgram)
+                {
+                    Utils.WriteSuccess(TracerProgramFound(datadog64BitProgram));
+                }
+                else if (displayName is datadog32BitProgram)
+                {
+                    if (RuntimeInformation.OSArchitecture is Architecture.X64)
+                    {
+                        Utils.WriteError(WrongTracerArchitecture(datadog32BitProgram));
+                        return tracerVersion;
+                    }
+
+                    Utils.WriteSuccess(TracerProgramFound(datadog32BitProgram));
+                }
+            }
+
+            return tracerVersion;
+        }
+
+        private static void CheckAppPools(ApplicationPool? pool)
+        {
+            var tracingRelevantVariables = new Dictionary<string, string>
+            {
+                { "COR_ENABLE_PROFILING", "1" },
+                { "CORECLR_ENABLE_PROFILING", "1" },
+                { "COR_PROFILER", Utils.Profilerid },
+                { "CORECLR_PROFILER", Utils.Profilerid }
+            };
+
+            var variableFound = false;
+            var foundVariables = new Dictionary<string, object>();
+
+            if (pool != null)
+            {
+                var environmentVariablesCollection = pool.GetCollection("environmentVariables");
+                foreach (var variable in environmentVariablesCollection)
+                {
+                    var name = (string)variable.Attributes["name"].Value;
+                    var value = (string)variable.Attributes["value"].Value;
+
+                    if (tracingRelevantVariables.ContainsKey(name) && value != tracingRelevantVariables[name])
+                    {
+                        variableFound = true;
+                        foundVariables[name] = value;
+                    }
+                }
+
+                if (variableFound)
+                {
+                    Utils.WriteWarning(AppPoolCheckFindings(pool.Name));
+
+                    foreach (var variable in foundVariables)
+                    {
+                        Utils.WriteError(WrongEnvironmentVariableFormat(variable.Key, tracingRelevantVariables[variable.Key], variable.Value.ToString()));
```

Comment:
Can `variable.Value` ever be `null`? 🤔  I'm not sure tbh, if so you can do `variable.Value?.ToString()`

Reply:
So for this one I did a test to check if an empty value could be saved or null but it seems they can't save the env var without a value, I did the test below to confirm:
![image](https://github.com/DataDog/dd-trace-dotnet/assets/28125428/710ea1d5-331e-422c-b0a1-65e6629f4f3f)

Also removed the `value` from the `applicationHost.config` file, and left it with an empty string as a value and this causes the service to not even run so:
![image](https://github.com/DataDog/dd-trace-dotnet/assets/28125428/c4b7f7a4-6282-4df8-b007-a29d228417ba)
![image](https://github.com/DataDog/dd-trace-dotnet/assets/28125428/65e5be3d-b468-4819-b4ba-e325ced03b4e)

In the code I get the line below when I tried to add the null check, so I wanted to understand better if even in this scenarios we should add it for possible runtime issues that could lead to the value being null:
`Conditional access qualifier expression is never null according to nullable reference types' annotations`

Reply:
Nah, that all looks good!